pkg/cloudprovider/providers/openstack/openstack.go

```diff
@@ -535,3 +539,96 @@ func (os *OpenStack) Routes() (cloudprovider.Routes, bool) {
 
 	return r, true
 }
+
+// Implementation of sort interface for blockstorage version probing
+type APIVersionsByID []apiversions_v1.APIVersion
+
+func (apiVersions APIVersionsByID) Len() int {
+	return len(apiVersions)
+}
+
+func (apiVersions APIVersionsByID) Swap(i, j int) {
+	apiVersions[i], apiVersions[j] = apiVersions[j], apiVersions[i]
+}
+
+func (apiVersions APIVersionsByID) Less(i, j int) bool {
+	return apiVersions[i].ID > apiVersions[j].ID
+}
+
+func (os *OpenStack) volumeService() (volumeService, error) {
+	switch os.bsOpts.BSVersion {
+	case "v1":
+		sClient, err := openstack.NewBlockStorageV1(os.provider, gophercloud.EndpointOpts{
+			Region: os.region,
+		})
+		if err != nil || sClient == nil {
+			glog.Errorf("Unable to initialize cinder client for region: %s", os.region)
+			return nil, err
+		}
+		return &VolumesV1{sClient, os.bsOpts}, nil
+	case "v2":
+		sClient, err := openstack.NewBlockStorageV2(os.provider, gophercloud.EndpointOpts{
+			Region: os.region,
+		})
+		if err != nil || sClient == nil {
+			glog.Errorf("Unable to initialize cinder v2 client for region: %s", os.region)
+			return nil, err
+		}
+		return &VolumesV2{sClient, os.bsOpts}, nil
+	case "auto":
+		sClient, err := openstack.NewBlockStorageV1(os.provider, gophercloud.EndpointOpts{
+			Region: os.region,
+		})
+		if err != nil || sClient == nil {
+			glog.Errorf("Unable to initialize cinder client for region: %s", os.region)
+			return nil, err
+		}
+
+		err = apiversions_v1.List(sClient).EachPage(func(page pagination.Page) (bool, error) {
+			apiversions, err := apiversions_v1.ExtractAPIVersions(page)
+			if err != nil {
+				glog.Errorf("Unable to list blockstorage API versions for region: %s", os.region)
+				return false, err
+			}
+
+			sort.Sort(APIVersionsByID(apiversions))
+			// take first CURRENT or SUPPORTED version
+			for _, version := range apiversions {
```

Comment:
apiversions is a []APIVersion, I assumed in order iteration

Can you post the output of your api version query? 
https://developer.openstack.org/api-ref/block-storage/v2/index.html?expanded=list-api-versions-detail#list-api-versions

I'll do unit tests for it.